syntax = "proto3";
import "google/protobuf/empty.proto";

package ebs;

service Server {
  rpc read (ReadReq) returns (ReadReply) {}
  rpc write (WriteReq) returns (WriteReply) {}
}

message ReadReq {
  uint64 offset = 1;
}

message ReadReply {
  bytes data = 1;
  int32 status = 2;
  string primary = 3;
}

message WriteReq {
  uint64 offset = 1;
  bytes data = 2;
}

message WriteReply {
  int32 status = 1;
  string primary = 2;
}

service Backup {
  rpc heartBeat(google.protobuf.Empty) returns (google.protobuf.Empty);
  rpc write(WriteReq) returns (WriteReply);
  rpc replayLog(ReplayReq) returns(ReplayReply);
}

message ReplayReq {
  repeated WriteReq item = 1;
}

message ReplayReply {
  int32 status = 1;
}
